### ISSUE TITLE
Consideration of multiple delimiters for sorting scripts

### ISSUE THREAD

**Issue by ofk** [cameronhunter/prettier-package-json#65]
There is a task runner package called [npm-run-all](https://www.npmjs.com/package/npm-run-all) which is used in the development of prettier, among other things.
This task runner uses `:` and `/` as delimiters for script names ([ref](https://github.com/mysticatea/npm-run-all/blob/HEAD/docs/npm-run-all.md#glob-like-pattern-matching-for-script-names)).

As a result, scripts are often written like this:
```json
{
  "scripts": {
    "lint": "npm-run-all lint:*",
    "lint:eslint": "eslint .",
    "lint:prettier": "prettier . --check",
    "lint-fix": "npm-run-all lint-fix:*",
    "lint-fix:eslint": "eslint . --fix",
    "lint-fix:prettier": "prettier . --write"
  }
}
```

However, prettier-package-json does not consider these delimiters when sorting, resulting in the following order:
```json
{
  "scripts": {
    "lint": "npm-run-all lint:*",
    "lint-fix": "npm-run-all lint-fix:*",
    "lint-fix:eslint": "eslint . --fix",
    "lint-fix:prettier": "prettier . --write",
    "lint:eslint": "eslint .",
    "lint:prettier": "prettier . --check"
  }
}
```

Are there plans to add this sort of consideration, or would a pull request for this be accepted?

For reference I am considering the following sorting:
```js
const scripts='lint lint-fix lint-fix:eslint lint-fix:prettier lint:eslint lint:prettier'.split(' ');

scripts.sort((a, b) => {
  const aa = a.split(/([:/])/);
  const bb = b.split(/([:/])/);
  const al = aa.length;
  const bl = bb.length;
  for (let i = 0; i < Math.min(al, bl); i++) {
    if (aa[i] !== bb[i]) {
      return aa[i] < bb[i] ? -1 : 1;
    }
  }
  if (al !== bl) {
    return al < bl ? -1 : 1;
  }
  return 0;
});

console.log(scripts); // => ['lint', 'lint:eslint', 'lint:prettier', 'lint-fix', 'lint-fix:eslint', 'lint-fix:prettier']
```

**Comment by ofk** [comment]
I am considering the following sorting:
```js
const scripts='lint lint-fix lint-fix:eslint lint-fix:prettier lint:eslint lint:prettier'.split(' ');

scripts.sort((a, b) => {
  const aa = a.split(/([:/])/);
  const bb = b.split(/([:/])/);
  const al = aa.length;
  const bl = bb.length;
  for (let i = 0; i < Math.min(al, bl); i++) {
    if (aa[i] !== bb[i]) {
      return aa[i] < bb[i] ? -1 : 1;
    }
  }
  if (al !== bl) {
    return al < bl ? -1 : 1;
  }
  return 0;
});

console.log(scripts); // => ['lint', 'lint:eslint', 'lint:prettier', 'lint-fix', 'lint-fix:eslint', 'lint-fix:prettier']
```

**Comment by cameronhunter** [comment]
Hi @ofk , this sounds good to me. A pull-request would be 👍. Thanks for contributing!

**Comment by cameronhunter** [comment]
Resolved in `v2.8.0`
